src/Serialise/Cborg/Tutorial.hs

```diff
@@ -7,42 +9,314 @@
+>       (2, 2) -> SwimmingAnimal <$> decode
+>       _      -> fail "invalid Animal encoding"
+
+=== Adding\/removing\/modifying fields
+
+Say then we want to add a new field to our @WalkingAnimal@ constructor,
+
+> data Animal = HoppingAnimal { animalName :: String, hoppingHeight :: Int }
+>             | WalkingAnimal { animalName :: String, walkingSpeed :: Int, numberOfFeet :: Int }
+>             | SwimmingAnimal { numberOfFins :: Int }
+>             deriving (Generic)
+
+We can account for this by representing @WalkingAnimal@ with a new encoding with
+a new tag,
+
+> encodeAnimal :: Animal -> Encoding
+> -- HoppingAnimal, SwimmingAnimal cases are unchanged...
+> encodeAnimal (HoppingAnimal name height) =
+>     encodeListLen 3 <> encodeTag 0 <> encode name <> encode height
+> encodeAnimal (WalkingAnimal name speed) =
+>     encodeListLen 3 <> encodeTag 1 <> encode name <> encode speed
+> -- This is new...
+> encodeAnimal (WalkingAnimal animalName walkingSpeed numberOfFeet) =
+>     encodeListLen 4 <> encodeTag 3 <> encode animalName <> encode walkingSpeed <> encode numberOfFins
+>
+> decodeAnimal :: Decoder s Animal
+> decodeAnimal = do
+>     len <- decodeListLen
+>     tag <- decodeTag
+>     case (len, tag) of
+>       -- this cases are unchanged...
+>       (3, 0) -> HoppingAnimal <$> decode <*> decode
+>       (2, 2) -> SwimmingAnimal <$> decode
+>       -- this is new...
+>       (3, 1) -> WalkingAnimal <$> decode <*> decode <*> pure 4
+>                                                      -- ^ note the default for backwards compat
+>       (4, 3) -> WalkingAnimal <$> decode <*> decode
+>       _      -> fail "invalid Animal encoding"
+
+We can use this same approach to handle field removal and type changes.
+
+-}
+
+{- $arbitrary_terms
+
+When working with foreign encodings, it can sometimes be useful to capture a
+serialised CBOR term verbatim (for instance, so you can later re-serialize it in
+some later result). The 'Serialise.Cborg.Term.Term' type provides such a
+representation, losslessly capturing a CBOR AST. It can be serialised and
+deserialised with its 'Serialise' instance.
+
+-}
+
+{- $examining_encodings
+
+We can also look In addition to serialisation and deserialisation, @cborg@
+provides a variety of tools for representing arbitrary CBOR encodings in the
+"Serialise.Cborg.FlatTerm" and "Serialise.Cborg.Pretty" modules.
+
+The 'Serialise.Cborg.FlatTerm.FlatTerm' type represents a single CBOR /term/, as
+would be found in the ultimate CBOR representation. For instance, we can easily
+look at the structure of our @Animal@ encoding above,
+
+>>> toFlatTerm $ encode $ HoppingAnimal "Fred" 42
+[TkListLen 3,TkInt 0,TkString "Fred",TkInt 42]
+>>> fromFlatTerm (decode @Animal) $ toFlatTerm $ encode (HoppingAnimal "Fred" 42)
+Right (HoppingAnimal {animalName = "Fred", hoppingHeight = 42})
+
+This can be useful both for understanding external CBOR formats, as well as
+understanding and testing your own hand-rolled encodings.
```

Comment:
Perhaps add a comment about the pretty printing of the encoded format too, to help debug things? (I realise these suggestions could go on forever...)